python/meep.i

```diff
@@ -841,7 +841,7 @@ void _get_gradient(PyObject *grad, PyObject *fields_a, PyObject *fields_f, PyObj
     if (!PyArray_ISCARRAY(pao_freqs)) meep::abort("Numpy fields array must be C-style contiguous.");
     double *frequencies_c = (double *)PyArray_DATA(pao_freqs);
     int nf = PyArray_DIMS(pao_freqs)[0];
-    if (PyArray_DIMS(pao_grad)[0] != nf) meep::abort("Numpy grad array is allocated for %d frequencies; it should be allocated for %d.",PyArray_DIMS(pao_grad)[0],nf);
+    if (PyArray_DIMS(pao_grad)[0] != nf) meep::abort("Numpy grad array is allocated for %ld frequencies; it should be allocated for %ld.",PyArray_DIMS(pao_grad)[0],nf);
```

Comment:
This still doesn't seem quite correct.  `PyArray_DIMS` returns a `npy_intp`, which is an "integer the same size as a pointer", similar to `ptrdiff_t`.  On a 64-bit Windows machine, however `%ld` (`long int`) is only 4 bytes, I think?  The standard C formatting code for this is `%td`, I think, but I'm not sure what is standard in C++?

Reply:
Updated to %td